cmd/network-slirp-binding/domain/domain.go

```diff
@@ -0,0 +1,175 @@
+	network := vmispec.LookupPodNetwork(networks)
+	if network == nil {
+		return nil, fmt.Errorf("pod network not found")
+	}
+	iface := vmispec.LookupInterfaceByName(ifaces, network.Name)
+	if iface == nil || iface.Slirp == nil {
+		return nil, fmt.Errorf("no slirp iface found")
+	}
+
+	return &SlirpNetworkConfigurator{
+		vmiSpecIface:   iface,
+		vmiSpecNetwork: network,
+		searchDomains:  searchDomains,
+	}, nil
+}
+
+func (s SlirpNetworkConfigurator) Mutate(domainSpec *domainschema.DomainSpec) (*domainschema.DomainSpec, error) {
+	slirpQemuCmdArgs, err := generateSlirpNetworkQemuCmdArgs(s.vmiSpecIface, s.vmiSpecNetwork, s.searchDomains)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(slirpQemuCmdArgs) == 0 {
+		log.Log.Warning("no qemu cmd args configured, domain spec did not change")
+		return domainSpec, nil
+	}
+
+	domainSpec.QEMUCmd = initQemuCmdArgs(domainSpec.QEMUCmd)
+	domainSpec.QEMUCmd.QEMUArg = append(domainSpec.QEMUCmd.QEMUArg, slirpQemuCmdArgs...)
+
+	return domainSpec, nil
+}
+
+func generateSlirpNetworkQemuCmdArgs(iface *vmschema.Interface, network *vmschema.Network, searchDomains []string) ([]domainschema.Arg, error) {
+	backendDeviceQemuArgs, err := generateBackendDeviceQemuArg(iface, network, searchDomains)
+	if err != nil {
+		return nil, err
+	}
+
+	var qemuArgs []domainschema.Arg
+	qemuArgs = append(qemuArgs, backendDeviceQemuArgs...)
+	qemuArgs = append(qemuArgs, generateNetDeviceQemuArg(iface)...)
+
+	return qemuArgs, nil
+}
+
+func generateBackendDeviceQemuArg(iface *vmschema.Interface, network *vmschema.Network, dnsDomains []string) ([]domainschema.Arg, error) {
+	backendDeviceConf := domainschema.Arg{Value: fmt.Sprintf("user,id=%s", iface.Name)}
+
+	cidr, err := networkCIDR(network)
+	if err != nil {
+		return nil, err
+	}
+	backendDeviceConf.Value += fmt.Sprintf(",net=%s", cidr)
+
+	for _, dnsDomain := range dnsDomains {
+		backendDeviceConf.Value += fmt.Sprintf(",dnssearch=%s", dnsDomain)
+	}
+
+	ports, err := ifacePorts(iface)
+	if err != nil {
+		return nil, err
+	}
+	for _, port := range ports {
+		backendDeviceConf.Value += fmt.Sprintf(",hostfwd=%[1]s::%[2]d-:%[2]d", strings.ToLower(port.Protocol), port.Port)
+	}
+
+	return []domainschema.Arg{{Value: "-netdev"}, backendDeviceConf}, nil
+}
+
+func networkCIDR(network *vmschema.Network) (string, error) {
+	if network.Pod.VMNetworkCIDR != "" {
+		if _, _, err := net.ParseCIDR(network.Pod.VMNetworkCIDR); err != nil {
+			return "", fmt.Errorf("invalid network CIDR %q: %v", network.Pod.VMNetworkCIDR, err)
+		}
+		return network.Pod.VMNetworkCIDR, nil
+	}
+
+	return domainschema.DefaultVMCIDR, nil
+}
+
+func ifacePorts(iface *vmschema.Interface) ([]vmschema.Port, error) {
+	if len(iface.Ports) == 0 {
+		return nil, nil
+	}
+
+	var ports []vmschema.Port
+	configuredPorts := make(map[string]struct{}, 0)
+	for _, ifacePort := range iface.Ports {
+		if ifacePort.Port <= 0 || ifacePort.Port >= 65536 {
```

Comment:
Pending feedback.

Reply:
The original implementation had few flaws, this is one of them although I tried to avoid changing the implementation as much as possible.
If its correct I think we should leave it, if you insist I will remove the upper limit.

Reply:
I removed the upper limit for port as in the current implementation.